src/libs/actions/Travel.ts

```diff
@@ -1,59 +1,168 @@
-                .then((response) => {
-                    if (!response?.spotnanaToken) {
-                        reject(error);
-                        throw error;
-                    }
-                    return buildTravelDotURL(response.spotnanaToken);
-                })
-                .catch(() => {
-                    reject(error);
-                    throw error;
-                }),
-            (travelDotURL) => {
-                resolve(travelDotURL);
-                return travelDotURL ?? '';
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
+            value: {
+                isLoading: false,
+                errors: getMicroSecondOnyxErrorWithTranslationKey('travel.errorMessage'),
             },
-        );
-    });
+        },
+    ];
+
+    const params: AcceptSpotnanaTermsParams = {domain};
+
+    API.write(WRITE_COMMANDS.ACCEPT_SPOTNANA_TERMS, params, {optimisticData, successData, failureData});
+}
+
+function handleProvisioningPermissionDeniedError(domain: string) {
+    Navigation.navigate(ROUTES.TRAVEL_DOMAIN_PERMISSION_INFO.getRoute(domain));
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+}
+
+function openTravelDotAfterProvisioning(spotnanaToken: string) {
+    Navigation.goBack();
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+    Linking.openURL(buildTravelDotURL(spotnanaToken));
+}
+
+function provisionDomain(domain: string) {
+    Navigation.goBack();
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+    Navigation.navigate(ROUTES.TRAVEL_TCS.getRoute(domain));
+}
+
+function bookATrip(translate: LocaleContextProps['translate'], setCtaErrorMessage: Dispatch<SetStateAction<string>>, ctaErrorMessage = ''): void {
+    if (!activePolicyID) {
+        return;
+    }
+    if (Str.isSMSLogin(primaryLogin)) {
+        setCtaErrorMessage(translate('travel.phoneError'));
+        return;
+    }
+    const policy = getPolicy(activePolicyID);
+    if (isEmptyObject(policy?.address)) {
+        Navigation.navigate(ROUTES.WORKSPACE_PROFILE_ADDRESS.getRoute(activePolicyID, Navigation.getActiveRoute()));
+        return;
+    }
+
+    const isPolicyProvisioned = policy?.travelSettings?.spotnanaCompanyID ?? policy?.travelSettings?.associatedTravelDomainAccountID;
+    if (policy?.travelSettings?.hasAcceptedTerms ?? (travelSettings?.hasAcceptedTerms && isPolicyProvisioned)) {
+        openTravelDotLink(activePolicyID)
+            ?.then(() => {
+                if (!NativeModules.HybridAppModule || !isSingleNewDotEntry) {
+                    return;
+                }
+
+                Log.info('[HybridApp] Returning to OldDot after opening TravelDot');
+                NativeModules.HybridAppModule.closeReactNativeApp(false, false);
+            })
+            ?.catch(() => {
+                setCtaErrorMessage(translate('travel.errorMessage'));
```

Comment:
If `Cta` is the acronym for "call to action", then this should be uppercased like `setCTAErrorMessage`

Reply:
Created a follow-up [issue](https://github.com/Expensify/App/issues/55755) for this.